src/hmem.c

```diff
@@ -212,18 +212,8 @@ ofi_async_copy_hmem_iov_buf(enum fi_hmem_iface hmem_iface, uint64_t device,
 		return -FI_EINVAL;
 
 	for (i = 0; i < hmem_iov_count && size; i++) {
-		len = hmem_iov[i].iov_len;
-
-		if (hmem_iov_offset > len) {
-			hmem_iov_offset -= len;
-			continue;
-		}
-
-		hmem_buf = (char *)hmem_iov[i].iov_base + hmem_iov_offset;
-		len -= hmem_iov_offset;
-		hmem_iov_offset = 0;
-
-		len = MIN(len, size);
```

Comment:
I understand this is a refactor. But why do we used to set `hmem_iov_offset` as 0 considering it is not used anymore in this function?

Reply:
never mind. I get it now. It is in a loop over the iov so it is still used.